CONTRIBUTING.md

```diff
@@ -72,11 +116,59 @@ See [here](guidelines/python-cli-ux-guidelines.md) for full UX guidelines to fol
 1. Search the [open issues](../../issues) in GitHub to find out what has been planned
 2. Select an existing issue or open an issue to propose changes or fixes
 3. Add any relevant labels as you work on it
-4. Run tests as described [in the main README](https://github.com/conjurinc/conjur-api-python3#testing),
-ensuring they pass
+4. Run tests as described in the [testing section of this document](https://github.com/cyberark/conjur-api-python3/blob/master/CONTRIBUTING.md#testing), ensuring they pass
 5. Submit a pull request, linking the issue in the description
 6. Adjust labels as-needed on the issue. Ask another contributor to review and merge your code if there are delays in merging.
 
 ## Releasing
 
-TODO: Define release workflow once we have a PyPI publishing in place
+To create a tag and release follow the instructions in this section.
+
+### Update the version, changelog, and notices
+
+1. Create a new branch for the version bump
+
+1. Based on the unreleased content, determine the new version number and update the version in `version.py`
+
+1. Review the git log and ensure the changelog contains all relevant recent changes with references to GitHub issues or PRs, if possible
+
+1. Review the changes since the last tag, and if the dependencies have changed revise the [notices](NOTICES.txt) file to correctly capture the added dependencies and their licenses / copyrights
+
+1. Before creating a release, ensure that all documentation that needs to be written has been written by TW, approved by PO/Engineer, and pushed to the forward-facing documentation
+
+1. Commit these changes to the branch. Bump version to x.y.z is an acceptable commit message and open a PR for review
+
+Add a git tag
+
+1. Once your changes have been reviewed and merged into master, tag the version using `git tag -s v0.1.1`. Note this requires you to be able to sign releases. Consult the [github documentation on signing commits](https://docs.github.com/en/free-pro-team@latest/github/authenticating-to-github/managing-commit-signature-verification) on how to set this up vx.y.z is an acceptable tag message
+1. Push the tag: `git push vx.y.z` (or `git push origin vx.y.z` if you are working from your local machine)
+
+### Add release artifacts
+
+Currently, packing the client into an executable is a manual process. For Linux and Windows, you must pack the client using the different VMs we have available to us. For macOS, you will need to use your local machine.
+
+#### Linux
+
+1. Copy `conjur`, `pkg_conjur`, and `requirements.txt` folders from your local machine to the Linux machine
```

Comment:
This may be better to be written as the following for all OS types:
- Clone the git repo on the machine
- Activate venv
- Install dependencies
- run `./bin/build_binary -l`

Reply:
Oh nice. Much more clear will add

Reply:
@sgnn7 why do we need to install dependencies on the machine if the Dockerfile.test used when we run `./bin/build_binary -l` has the following already in it:

```
RUN apt-get update && \
    apt-get install -y bash \
                       binutils \
                       build-essential \
                       curl \
                       git \
                       jq \
                       libffi-dev \
                       libssl-dev \
                       python3 \
                       python3-dev \
                       python3-pip
```